### PR TITLE
Update dependency go modules for k8s v1.28.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.28.0
```release-note
Update kubernetes dependencies to v1.28.0
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,10 +9,10 @@ require (
 	github.com/prometheus/client_golang v1.16.0
 	github.com/prometheus/client_model v0.4.0
 	github.com/prometheus/common v0.44.0
-	k8s.io/api v0.28.0-rc.0
-	k8s.io/apimachinery v0.28.0-rc.0
-	k8s.io/client-go v0.28.0-rc.0
-	k8s.io/component-base v0.28.0-rc.0
+	k8s.io/api v0.28.0
```
Comment (Sneha-at) [inline go.mod:12]: Line 6 has `csi-lib-utils v0.12.0` need to update to `v0.15.0` when ready

File: go.mod [hunk 2]
```diff
@@ -62,5 +62,3 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
-
```
Comment (sunnylovestiramisu) [inline go.mod:65]: This should be kept here, and we need to update the dependency under the path client/ as well.

This is a tricky one.

Comment (Sneha-at) [reply go.mod]: Updated the dependencies for `./client` and `csi-lib-utils`.

**Comment by msau42** [comment]
Can you edit the release notes?

**Comment by Sneha-at** [comment]
@msau42 Updated the release note

**Pull request merged**